### PR TITLE
Kucoin balance

### PR DESCRIPTION
**Pull request by samgermain**
Related to https://github.com/ccxt/ccxt/issues/12320

- Clean up 
- unifies the param name code in `kucoinfutures.fetchBalance` in params
- defaultCode is 'USDT'

--------------

```
% kucoinfutures fetchBalance '{"code": "BTC"}'          % kucoinfutures fetchBalance                   
2022-03-16T00:31:14.070Z                                2022-03-16T00:31:21.087Z
Node.js: v14.17.0                                       Node.js: v14.17.0
CCXT v1.75.93                                           CCXT v1.75.93
kucoinfutures.fetchBalance ([object Object])            kucoinfutures.fetchBalance ()
2022-03-16T00:31:14.648Z iteration 0 passed in 256 ms   2022-03-16T00:31:22.270Z iteration 0 passed in 666 ms
                                                        
{                                                       {
  info: {                                                 info: {
    code: '200000',                                         code: '200000',
    data: {                                                 data: {
      accountEquity: 0,                                       accountEquity: 9.9984053112,
      unrealisedPNL: 0,                                       unrealisedPNL: 0.003,
      marginBalance: 0,                                       marginBalance: 9.9954053112,
      positionMargin: 0,                                      positionMargin: 7.6701896,
      orderMargin: 0,                                         orderMargin: 0,
      frozenFunds: 0,                                         frozenFunds: 0,
      availableBalance: 0,                                    availableBalance: 2.3282157112,
      currency: 'XBT'                                         currency: 'USDT'
    }                                                       }
  },                                                      },
  timestamp: undefined,                                   timestamp: undefined,
  datetime: undefined,                                    datetime: undefined,
  BTC: { free: 0, used: 0, total: 0 },                    USDT: { free: 2.3282157112, used: 7.6701896, total: 9.9984053112 },
  free: { BTC: 0 },                                       free: { USDT: 2.3282157112 },
  used: { BTC: 0 },                                       used: { USDT: 7.6701896 },
  total: { BTC: 0 }                                       total: { USDT: 9.9984053112 }
}                                                       }
```

### CODE REVIEW

File: js/kucoinfutures.js [hunk 1]
```diff
@@ -1256,14 +1253,9 @@ module.exports = class kucoinfutures extends kucoin {
...
-        let defaultCode = this.safeString (this.options, 'code');
-        const fetchBalanceOptions = this.safeValue (this.options, 'fetchBalance', {});
-        defaultCode = this.safeString (fetchBalanceOptions, 'code', defaultCode);
-        const code = this.safeString (params, 'code', defaultCode);
-        if (coin !== undefined) {
-            request['currency'] = coin;
-        } else if (code !== undefined) {
+        const defaultCode = this.safeString (this.options, 'defaultCode');
+        const code = this.safeString2 (params, 'code', 'currency', defaultCode);
+        if (code !== undefined) {
```
Comment (kroitor) [inline js/kucoinfutures.js:1258]: We used to support both the exchange-specific param for the currency id and the unified currency code separately cause we should not mix them.

File: js/kucoinfutures.js [hunk 2]
```diff
@@ -1253,9 +1256,14 @@ module.exports = class kucoinfutures extends kucoin {
         await this.loadMarkets ();
         // only fetches one balance at a time
         const request = {};
-        const defaultCode = this.safeString (this.options, 'defaultCode');
-        const code = this.safeString2 (params, 'code', 'currency', defaultCode);
-        if (code !== undefined) {
+        const currencyId = this.safeString (params, 'currency');
+        let defaultCode = this.safeString (this.options, 'code');
+        const fetchBalanceOptions = this.safeValue (this.options, 'fetchBalance', {});
```
Comment (samgermain) [inline js/kucoinfutures.js:1262]: Whats the point of doing stuff like this? fetchBalance isn't in options

File: js/kucoinfutures.js [hunk 3]
```diff
@@ -1253,9 +1256,14 @@ module.exports = class kucoinfutures extends kucoin {
         await this.loadMarkets ();
         // only fetches one balance at a time
         const request = {};
-        const defaultCode = this.safeString (this.options, 'defaultCode');
-        const code = this.safeString2 (params, 'code', 'currency', defaultCode);
-        if (code !== undefined) {
+        const currencyId = this.safeString (params, 'currency');
+        let defaultCode = this.safeString (this.options, 'code');
```
Comment (samgermain) [inline js/kucoinfutures.js:1261]: I think that it should be `defaultCode`, it's more obvious to the user what that would be

File: js/kucoinfutures.js [hunk 4]
```diff
@@ -268,7 +268,6 @@ module.exports = class kucoinfutures extends kucoin {
                 'version': 'v1',
                 'symbolSeparator': '-',
                 'defaultType': 'swap',
-                'defaultCode': 'USDT',
```
Comment (samgermain) [inline js/kucoinfutures.js]: I am intentionally setting the default to `'USDT'`

**Comment by samgermain** [comment]
@kroitor I have few questions as comments on this PR

https://github.com/ccxt/ccxt/pull/12353#discussion_r827543619
https://github.com/ccxt/ccxt/pull/12353#discussion_r827544460
https://github.com/ccxt/ccxt/pull/12353#discussion_r827551688

In the future can we make sure that all the comments are resolved before a PR is merged?

**Pull request merged**